coq-core.opam

```diff
@@ -0,0 +1,34 @@
+synopsis: "The Coq Proof Assistant"
+description: """
+Coq is a formal proof management system. It provides
+a formal language to write mathematical definitions, executable
+algorithms and theorems together with an environment for
+semi-interactive development of machine-checked proofs. Typical
+applications include the certification of properties of programming
+languages (e.g. the CompCert compiler certification project, or the
+Bedrock verified low-level programming library), the formalization of
+mathematics (e.g. the full formalization of the Feit-Thompson theorem
+or homotopy type theory) and teaching.
+"""
+opam-version: "2.0"
+maintainer: "The Coq development team <coqdev@inria.fr>"
+authors: "The Coq development team, INRIA, CNRS, and contributors."
+homepage: "https://coq.inria.fr/"
+bug-reports: "https://github.com/coq/coq/issues"
+dev-repo: "git+https://github.com/coq/coq.git"
+license: "LGPL-2.1"
+
+version: "dev"
+
+depends: [
+  "ocaml"     { >= "4.05.0" }
+  "dune"      { >= "2.5.0"  }
+  "ocamlfind" { build }
+  "num"
```

Comment:
Why was a dependency on `num` added?

Also, should the dependency on `dune` be declared `build` only? I do not understand why users should be forced to recompile Coq (and all the depending Coq libraries) whenever one updates Dune. (I am pretty sure I will never update Dune if I know it will cause the recompilation of all the Coq libraries.)

Reply:
> Also, should the dependency on `dune` be declared `build` only? I do not understand why users should be forced to recompile Coq (and all the depending Coq libraries) whenever one updates Dune. (I am pretty sure I will never update Dune if I know it will cause the recompilation of all the Coq libraries.)

IIRC Dune installs `dune-project` files which contains the Dune version. Dune tries as hard as it can to be backward-compatible but if you downgrade your Dune version after having installed some Dune package, then you break your installation. Anyway, this led to removing the `{ build }` annotation from every `dune` dependency in the opam-repository.

Reply:
> Dune installs `dune-project` files which contains the Dune version.

Did you mean `dune-package`? Also, as far as I can tell, these files do not contain the version of Dune (but they contain the language version, which seems sensible to me).

> if you downgrade your Dune version after having installed some Dune package, then you break your installation

I have a hard time understanding why "downgrading" is a scenario worth making the life of every user painful. So yes, I guess I will never upgrade Dune, if it forces me to recompile every single package.

Reply:
Right. I don't know how frequently this happens. You should have a look at the relevant issues in the opam-repository for the context (I don't have the pointers, sorry).

Reply:
`num` seems like a bad rebase, thanks for catching it [I didn't do myself a super careful review until I fix the CI first]

`dune` cannot be `build` these days indeed, actually there is a very long discussion about that, as both dune-packages files and build actually are problematic, people is aware of it and trying to find a solution [I guess for dune 3.0] I suggest to discuss in the proper channel.

Note that if you enable compilation cache, and you upgrade dune, it should be expected that the recompilation time for packages building with dune to be close to zero. Not that this is a solution, but indeed changes the point of view a bit when build rules are effectively memoized.

Reply:
Resolved, that was a stale thing as we had to disable opam file generation.